mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1535,6 +1541,57 @@ struct GridwiseAttentionAccelRewritePattern
     return rewriter.create<TransformOp>(loc, operand, trMap);
   }
 
+  /// Check whether the op can bypass LDS-based swizzling
+  /// for the B operand of the second gemm.
+  bool canBypassLDSForSecondGemm(GridwiseAttentionAccelOp op) const {
+    Type elemTypeQ =
+        op.getQueries().getType().cast<MemRefType>().getElementType();
+    Type elemTypeK = op.getKeys().getType().cast<MemRefType>().getElementType();
+    StringRef arch = op.getArch();
+    RockAccelTuningParamAttrInterface gemm0TuningParams = op.getParams0();
+    auto accelEmitterPtrGemm0 = rock::accel::AccelEmitter::select(
+        op.getFeatures(), elemTypeQ, elemTypeK, arch, gemm0TuningParams);
+    bool isMFMA = bitEnumContainsAny(op.getFeatures(), GemmFeatures::mfma);
+    if (isMFMA) {
+      auto mfmaEmitter =
+          static_cast<rock::accel::MfmaEmitter *>(accelEmitterPtrGemm0.get());
```

Comment:
Instead of `static_cast` here, can we instead opt `MfmaEmitter` into the LLVM `cast<>` and `dyn_cast<>` infrastructure, please?

Reply:
ok, adopted LLVM RTTI into AccelEmitter and friends.